src/ldp/http/BasicResponseWriter.ts

```diff
@@ -20,8 +20,9 @@ export class BasicResponseWriter extends ResponseWriter {
     this.metadataWriter = metadataWriter;
   }
 
-  public async canHandle(input: { response: HttpResponse; result: ResponseDescription | Error }): Promise<void> {
-    if (isNativeError(input.result) || input.result.metadata?.contentType === INTERNAL_QUADS) {
+  public async canHandle(input: { response: HttpResponse; result: ResponseDescription }): Promise<void> {
+    const contentType = input.result.metadata?.contentType;
+    if (contentType && matchesMediaType(contentType, INTERNAL_ALL)) {
       throw new NotImplementedHttpError('Only successful binary responses are supported');
```

Comment:
Let's instead report on what is actually happening (because text is fine really).

```suggestion
      throw new NotImplementedHttpError(`Cannot serialize the internal content type ${contentType}`);
```